### ISSUE TITLE
Encoding Error on showing File Content

### ISSUE THREAD

**Issue by Azaret** [gitlabhq/gitlabhq#6672]
When opening some files on GitLab it shows Error 500 Encoding Error.
As far as I can tell there is not much more special carateres on the files it show the error than the ones which works fine.

Here the logs : 

```
Started GET "/csharp/*****/blob/master/Src/******/************.cs" for 192.168.10.77 at 2014-04-01 12:13:01 +0200
Processing by Projects::BlobController#show as HTML
Parameters: {"project_id"=>"csharp/******", "id"=>"master/Src/******/************.cs"}
Rendered shared/_ref_switcher.html.haml (6.2ms)
Rendered projects/commits/_commit.html.haml (8.8ms)
Rendered projects/blob/_actions.html.haml (6.4ms)
Rendered shared/_file_hljs.html.haml (153.9ms)
Rendered projects/blob/_text.html.haml (154.5ms)
Rendered projects/blob/_blob.html.haml (179.7ms)
Rendered projects/blob/show.html.haml within layouts/projects (188.4ms)
Encoding::CompatibilityError (incompatible character encodings: ASCII-8BIT and UTF-8):
app/views/projects/blob/show.html.haml:1:in `_app_views_projects_blob_show_html_haml__2623981704671252251_41845540'
app/controllers/application_controller.rb:58:in `set_current_user_for_thread'
Rendered errors/encoding.html.haml within layouts/errors (0.2ms)
Rendered layouts/_head.html.haml (1.8ms)
Rendered layouts/_search.html.haml (1.5ms)
Rendered layouts/_head_panel.html.haml (10.8ms)
Rendered layouts/_flash.html.haml (0.3ms)
Completed 500 Internal Server Error in 295ms (Views: 15.4ms | ActiveRecord: 4.1ms)
```

Not sure this is a duplicate or not, too much 500 errors tickets. So in the doub I post this. Sorry in that case.
Regards,

**Comment by DouweM** [comment]
A fix to this issue has been merged and will be in version 7.9. Please reopen this issue if the problem persists after upgrading to 7.9.

**Comment by Azaret** [comment]
It does work, thank a lot for the fix.
